src/components/SearchInput/SearchInput.types.ts

```diff
@@ -1,6 +1,8 @@
 import { CSSProperties } from 'react';
 import { AriaSearchFieldProps } from '@react-types/searchfield';
 
+export type SearchFieldHeight = 28 | 32;
```

Comment:
The problem is, you defined the height 2 places, here and in the `SearchInput.constants.ts`. You should have a single source of truth. You can reuse the `HEIGHT` constant here:

```suggestion
export type SearchFieldHeight = typeof HEIGHT[number];
```

This is not a perfect solution because we are importing a value into a type definition files, but from the TS point of view both just .ts files so ¯\_(ツ)_/¯

Another solution would be to convert union type to array type (`1 | 2 | 3` => `[1, 2, 3]`) but it involves some type magic... maybe next time